plugins/gatsby-source-data/main.js

```diff
@@ -127,53 +126,86 @@ const getValidTags = playlist => {
 
 const getPlaylistsFromYoutube = async (orgData, apiKey) => {
   const yt = new Youtube(apiKey);
-  let allPlaylistWithVideos = [];
-
-  for (org of orgData) {
-    if(org.youtubeId){
-      const channelPlaylists = await yt.getChannelPlaylists(org.youtubeId);
-      const playlistArr = [];
-      for (playlist of channelPlaylists) {
-        const tags = getValidTags(playlist);
-        // Only process the playlist if it has valid tags.
-        if(tags.category.length > 0){
-          const playlistObj = {
-            id: playlist.id,
-            title: playlist.snippet.title,
-            organisation: playlist.snippet.channelTitle,
-            donationMethod: "Paynow to UEN " + org.paynowUen,
-            language: tags.language.join(','),
-            tags: tags.category.join(','),
-            platform: "YouTube",
-            pageUrl: org.youtubeUrl || "",
-            thumbnailUrl: playlist.snippet.thumbnails.medium.url, // Encountered error: thumbnails key (i.e. thumbnails.standard) may not exist.
-          };
-
-          const playlistVideos = await yt.getPlaylistVideos(playlist.id);
-          const videosArr = playlistVideos.map(video => {
-            return {
-              id: video.snippet.resourceId.videoId,
-              playlistId: video.snippet.playlistId,
-              title: video.snippet.title,
-              asatizah: "<Asatizah name>", // TODO: Parse from description/title?
-              language: tags.language[0],  // TODO: Frontend needs to take an array instead.
-              addedOn: video.snippet.publishedAt,
-              videoUrl:
-                "https://www.youtube.com/embed/" + video.snippet.resourceId.videoId,
-            };
-          });
-
-          playlistObj.videos = videosArr;
-          playlistArr.push(playlistObj);
-        }
-      }
 
-      allPlaylistWithVideos = [...allPlaylistWithVideos, ...playlistArr];
-    }
-  }
+  // Only keep orgs with youtubeId
+  orgData = orgData.filter(org => !!org.youtubeId)
+
+  // Create array of promises to get playlists from org channels.
+  const getPlaylists = orgData.map(channel => yt.getChannelPlaylists(channel.youtubeId))
+
+  // Get all playlists on each channel.
+  let channelPlaylists = await Promise.all(getPlaylists)
+
+  // Add organisationName and donationUrl to channelPlaylists.
+  channelPlaylists = channelPlaylists.map(playlists => {
+    return playlists.map((playlist, index) => ({
+      ...playlist,
+      organisationName: orgData[index].organisationName,
+      donationUrl: orgData[index].donationUrl,
+    }))
+  })
+
+  // Flatten array of arrays. Now its a single array of playlists.
+  let playlists = channelPlaylists.flat(1)
```

Comment:
Since you're flattening after the `map`, you can combine it into a `reduce` (also to remove the use of `flat()` that is causing the build to fail):
```
channelPlaylists = channelPlaylists
  .reduce((acc, curr) => [...acc, ...curr], [])
  .map((playlist, index) => ({
    ...playlist,
    organisationName: orgData[index].organisationName,
    donationUrl: orgData[index].donationUrl,
  }))
```